wicket-core/src/main/java/org/apache/wicket/DefaultPageManagerProvider.java

```diff
@@ -167,6 +168,10 @@ protected IPageStore newPersistentStore()
 		Bytes maxSizePerSession = storeSettings.getMaxSizePerSession();
 		File fileStoreFolder = storeSettings.getFileStoreFolder();
 
-		return new DiskPageStore(application.getName(), fileStoreFolder, maxSizePerSession, getSerializer());
+		if (storeSettings.isEncrypted()) {
```

Comment:
Well the delegation could be done by a default implementation (for example as long as you return a store to delegate to). But I think you are right - the store configuration is rather static and because of this the stack is more verbose and not worth the effort to make it that configurable.